src/games/supported/Freeway.cpp

```diff
@@ -110,20 +109,16 @@ ModeVect FreewaySettings::getAvailableModes() {
 // the given mode must be one returned by the previous function
 void FreewaySettings::setMode(game_mode_t m, System &system,
                               std::unique_ptr<StellaEnvironmentWrapper> environment) {
-    if (m < getNumNodes()) { /*m >= 0 is implicit, since m is an unsigned int*/
-        m_mode = m;
-        // read the mode we are currently in
-        unsigned char mode = readRam(&system, 0x80);
-        // press select until the correct mode is reached
-        while (mode != m_mode) {
-            environment->pressSelect(1);
-            mode = readRam(&system, 0x80);
-        }
-        //reset the environment to apply changes.
-        environment->softReset();
-    } else {
-        throw std::runtime_error("This mode doesn't currently exist for this game");
+
+    // read the mode we are currently in
+    unsigned char mode = readRam(&system, 0x80);
+    // press select until the correct mode is reached
+    while (mode != m) {

```

Comment:
Was there a reason for removing the check `if (m < getNumModes())`? Currently this code will loop forever if an invalid m is given.

Reply:
Yes. I changed `ale_interface.cpp`. Before I'd only call `setMode` and that was it. Now it reads like this:

```
void ALEInterface::setMode(game_mode_t m) {
  //We first need to make sure m is an available mode
  ModeVect available = romSettings->getAvailableModes();
  if(find(available.begin(), available.end(), m) != available.end()) {
    environment->setMode(m);
  } else {
    throw std::runtime_error("This mode doesn't currently exist for this game");
  }
}
```

This guarantees that `environment->setMode(m)` will only be called if the mode passed as parameter is valid. Thus, I think that check I removed was redundant.